ports/esp32/machine_pwm.c

```diff
@@ -254,27 +246,21 @@ static void apply_duty(machine_pwm_obj_t *self) {
             .hpoint = 0,
             .flags.output_invert = self->output_invert,
         };
-        self->output_is_inverted = false;
-        if (self->channel_duty == max_timer_duty) {
-            cfg.duty = 0;
-            cfg.flags.output_invert = !self->output_invert;
-            self->output_is_inverted = true;
-        }
         check_esp_err(ledc_channel_config(&cfg));
         if (self->light_sleep_enable) {
             // Disable SLP_SEL to change GPIO status automantically in lightsleep.
             check_esp_err(gpio_sleep_sel_dis(self->pin));
             chans[self->mode][self->channel].light_sleep_enable = true;
         }
+        reconfigure_pin(self);
     } else {
-        #if FADE
```

Comment:
`#if FADE` seems better than `#ifdef FADE` . The latter is True even id FADE was defined as 0, while the former is False even when FADE is not defined. A complete pedantic expression would be `if defined(FADE) && FADE != 0`